cmd/cloud_test_.go

```diff
@@ -1,20 +1,584 @@
+		go func() {
+			stopErr := client.StopCloudTestRun(refID)
+			if stopErr != nil {
+				logger.WithError(stopErr).Error("Stop cloud test error")
+			} else {
+				logger.Info("Successfully sent signal to stop the cloud test, now waiting for it to actually stop...")
+			}
+			globalCancel()
+		}()
+	}
+	onHardStop := func(sig os.Signal) {
+		logger.WithField("sig", sig).Error("Aborting k6 in response to signal, we won't wait for the test to end.")
+	}
+	stopSignalHandling := handleTestAbortSignals(c.gs, gracefulStop, onHardStop)
+	defer stopSignalHandling()
+
+	cloudConfig.TestRunDetails = cloudTestRun.RuntimeConfig.TestRunDetails
+	testURL := cloudapi.URLForResults(refID, cloudConfig)
+	printExecutionDescriptionBasic(c.gs, testURL)
+
+	modifyAndPrintBar(
+		c.gs, progressBar,
+		pb.WithConstLeft("Run "), pb.WithConstProgress(0, "Initializing the cloud test"),
+	)
+
+	progressCtx, progressCancel := context.WithCancel(globalCtx)
+	progressBarWG := &sync.WaitGroup{}
+	progressBarWG.Add(1)
+	defer progressBarWG.Wait()
+	defer progressCancel()
+	go func() {
+		showProgress(progressCtx, c.gs, []*pb.ProgressBar{progressBar}, logger)
+		progressBarWG.Done()
+	}()
+
+	var (
+		startTime   time.Time
+		maxDuration time.Duration
+	)
+
+	testProgressLock := &sync.Mutex{}
+	var testProgress *cloudapi.TestProgressResponse
+	progressBar.Modify(
+		pb.WithProgress(func() (float64, []string) {
+			testProgressLock.Lock()
+			defer testProgressLock.Unlock()
+
+			if testProgress == nil {
+				return 0, []string{"Waiting..."}
+			}
+
+			statusText := testProgress.RunStatusText
+
+			if testProgress.RunStatus == cloudapi.RunStatusRunning {
+				if startTime.IsZero() {
+					startTime = time.Now()
+				}
+				spent := time.Since(startTime)
+				if spent > maxDuration {
+					statusText = maxDuration.String()
+				} else {
+					statusText = fmt.Sprintf("%s/%s", pb.GetFixedLengthDuration(spent, maxDuration), maxDuration)
+				}
+			} else if testProgress.RunStatus == cloudapi.RunStatusFinished {
+				testProgress.Progress = 1
+			}
+
+			return testProgress.Progress, []string{statusText}
+		}),
+	)
+
+	ticker := time.NewTicker(time.Millisecond * 2000)
+	if c.showCloudLogs {
+		go func() {
+			logger.Debug("Connecting to cloud logs server...")
+			if err := cloudConfig.StreamLogsToLogger(globalCtx, logger, refID, 0); err != nil {
+				logger.WithError(err).Error("error while tailing cloud logs")
+			}
+		}()
+	}
+
+	for range ticker.C {
+		newTestProgress, progressErr := client.GetTestProgress(refID)
+		if progressErr != nil {
+			logger.WithError(progressErr).Error("Test progress error")
+			continue
+		}
+		if testProgress != nil &&
+			testProgress.RunStatus != newTestProgress.RunStatus &&
+			newTestProgress.RunStatus == cloudapi.RunStatusRunning {
+
+			var cloudTestRun *cloudapi.CloudTestRun
+			cloudTestRun, err = apiClient.GetCloudTestRun(refID)
+			maxDuration = time.Duration(cloudTestRun.Duration * 1e9)
```

Comment:
Test duration value is available only after the test is validated, so I'm setting it here as part of test progress polling.